ibis/expr/types/pretty.py

```diff
@@ -219,21 +262,76 @@ def isnull(x):
     return out, min_width, max_width
 
 
-def format_dtype(dtype):
-    max_string = ibis.options.repr.interactive.max_string
+def format_dtype(dtype, max_string: int) -> Text:
     strtyp = str(dtype)
     if len(strtyp) > max_string:
         strtyp = strtyp[: max_string - 1] + "…"
     return Text.styled(strtyp, "dim")
 
 
-def to_rich_table(table, console_width=None):
-    if console_width is None:
-        console_width = float("inf")
+def to_rich(
```

Comment:
could use singledispatch, but that felt like overkill for two branches

Reply:
Yeah, this seems fine as you have it.